pcdsdevices/interface.py

```diff
@@ -0,0 +1,893 @@
+"""
+Module for defining bell-and-whistles movement features
+"""
+import time
+import fcntl
+import logging
+import numbers
+import signal
+import re
+from contextlib import contextmanager
+from pathlib import Path
+from threading import Thread, Event
+from types import SimpleNamespace, MethodType
+from weakref import WeakSet
+
+import yaml
+from bluesky.utils import ProgressBar
+from ophyd.device import Kind
+from ophyd.status import wait as status_wait
+
+from . import utils as util
+
+logger = logging.getLogger(__name__)
+engineering_mode = True
+
+OphydObject_whitelist = ["name", "connected", "check_value", "log"]
```

Comment:
Completely unnecessary to refactor, but a thought: `whitelist_classes = {OphydObj: ['whitelist', ], Device: [...]}` - a dictionary of class to whitelist, that could easily be summed below.

Reply:
Yes, this probably would have been a cleaner way to do it